multihash.go

```diff
@@ -285,12 +280,6 @@ func EncodeName(buf []byte, name string) ([]byte, error) {
 	return Encode(buf, Names[name])
 }
 
-// ValidCode checks whether a multihash code is valid.
-func ValidCode(code uint64) bool {
```

Comment:
known code?

Reply:
I think we'd need to clarify what this function is _for_, if we reintroduce it.

If I add back a function called `KnownCode`, and the body checks the new `registry` map for a implementation existing, and call it in all the places `ValidCode` was previously called... I get test failures (for [these](https://github.com/multiformats/go-multihash/blob/6f1ea18f1da5f7735ea31b5e2011da61c409e37f/multihash_test.go#L59-L60), though I think there may be... legitimate issues with those entries).)

If the idea is to check whether the number could describe a multihash at all (rather than whether we have an implementation registered for it), it would be better to implement it based on go-multicodec, but that recurses into #137 town, [which I no longer want to wrangle in this PR](https://github.com/multiformats/go-multihash/pull/136#discussion_r588875295) because the scope creep is slowing this down.

In either case, I think this function's core concept is questionable, because it swims against "be a data carrier" and "forward compatible" as an objective.

Reply:
I agree. I think "supported" may be useful (i.e., can we compute a hash with this code), but we can add that later.